### PR TITLE
Feature remote refactor

### PR DESCRIPTION
**Pull request by timbob1000**
Yeah, got a little scope creep on this one.  Here are some highlights:

Warnings to 0
Removed all 3rd party dependencies
Re-wrote the database server
Re-wrote the web server
Removed web server client
Got it working on Android
Improved performance by 2x 
Removed all memory leaks
Fixed a slew of bugs
Created a new index type build on hash table and hash matrix
Added more relevance to the loadFactor
Improved storage footprint

### CODE REVIEW

File: onyx-database-examples/lifecycle-events/src/main/java/com/onyxdevtools/lifecycle/AbstractDemo.java [hunk 1]
```diff
@@ -1,97 +1,17 @@
...
      * Helper method to assert a boolean
      * @param message Message to show if assert fails
      * @param operator boolean to assert
      */
-    public static void assertTrue(String message, boolean operator)
+    static void assertTrue(String message, boolean operator)
     {
         if(!operator)
         {
             System.err.println(message);
```
Comment (cosbor11) [inline onyx-database-examples/lifecycle-events/src/main/java/com/onyxdevtools/lifecycle/AbstractDemo.java]: replace with logging

File: onyx-database-examples/performance-benchmarks/pom.xml [hunk 2]
```diff
@@ -62,6 +62,15 @@
 
     <build>
         <plugins>
+
+            <plugin>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>1.8</source>
```
Comment (cosbor11) [inline onyx-database-examples/performance-benchmarks/pom.xml]: ${java.version}

File: onyx-database/pom.xml [hunk 3]
```diff
@@ -6,6 +6,7 @@
     <artifactId>onyx-database</artifactId>
     <name>${project.groupId}:${project.artifactId}</name>
     <description>Onyx Database</description>
+    <url>http://onyxdevtools.com</url>
```
Comment (cosbor11) [inline onyx-database/pom.xml]: https

File: onyx-database-examples/performance-benchmarks/pom.xml (hunk 2 above)
Comment (timbob1000) [inline onyx-database-examples/performance-benchmarks/pom.xml]: addressed in next commit

File: onyx-database/pom.xml (hunk 3 above)
Comment (timbob1000) [inline onyx-database/pom.xml]: addressed in next commit

File: onyx-database-examples/lifecycle-events/src/main/java/com/onyxdevtools/lifecycle/AbstractDemo.java (hunk 1 above)
Comment (timbob1000) [inline onyx-database-examples/lifecycle-events/src/main/java/com/onyxdevtools/lifecycle/AbstractDemo.java]: This is for an example project.  I do not want to obfuscate it with anything that does not have to do with Onyx

**Pull request merged**